src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -73,7 +73,7 @@ internal sealed class LocalFunctionSymbol : MethodSymbol
             }
 
             _binder = binder;
-            _refKind = (syntax.ReturnType.Kind() == SyntaxKind.RefType) ? RefKind.Ref : RefKind.None;
+            _syntax.ReturnType.SkipRef(out _refKind);
```

Comment:
we do not keep the syntax. we keep the reference, so it is not as trivial. I think I can make ref kind lazy for local functions. 
Not so sure about MemberMethod though - getting return type is complicated there because of overriding. I'll keep it eager for MemberMethod.

---
In reply to: [109756907](https://github.com/dotnet/roslyn/pull/18419#discussion_r109756907) [](ancestors = 109756907)